cardano-testnet/src/Testnet/Defaults.hs

```diff
@@ -418,14 +445,40 @@ defaultShelleyGenesis startTime testnetOptions =
       slotLength = cardanoSlotLength testnetOptions
       epochLength = cardanoEpochLength testnetOptions
       maxLovelaceLovelaceSupply = cardanoMaxSupply testnetOptions
+      pVer = eraToProtocolVersion $ cardanoNodeEra testnetOptions
+      protocolParams = Api.sgProtocolParams Api.shelleyGenesisDefaults
+      protocolParamsWithPVer = protocolParams & ppProtocolVersionL' .~ pVer
   in Api.shelleyGenesisDefaults
         { Api.sgNetworkMagic = fromIntegral testnetMagic
         , Api.sgSlotLength = secondsToNominalDiffTimeMicro $ realToFrac slotLength
         , Api.sgEpochLength = EpochSize $ fromIntegral epochLength
         , Api.sgMaxLovelaceSupply = maxLovelaceLovelaceSupply
         , Api.sgSystemStart = startTime
+        , Api.sgProtocolParams = protocolParamsWithPVer
         }
 
+
+eraToProtocolVersion :: AnyCardanoEra -> ProtVer
+eraToProtocolVersion (AnyCardanoEra era) =
+  case era of
+    ByronEra -> error "eraToProtocolVersion: Byron not supported"
+    ShelleyEra -> mkProtVer (3, 0)
+    AllegraEra -> mkProtVer (4, 0)
+    MaryEra -> mkProtVer (5, 0)
+    AlonzoEra -> mkProtVer (7, 2)
+    BabbageEra -> mkProtVer (8, 0)
+    ConwayEra -> mkProtVer (9, 0)
```

Comment:
Byron = 1
Shelley = 2
Allegra = 3
Mary = 4
Alonzo 5 and 6
Babbage 7 and 8
Conway 9